lib/spack/spack/build_systems/cached_cmake.py

```diff
@@ -193,7 +193,10 @@ def initconfig_mpi_entries(self):
 
         entries.append(cmake_cache_path("MPI_C_COMPILER", spec["mpi"].mpicc))
         entries.append(cmake_cache_path("MPI_CXX_COMPILER", spec["mpi"].mpicxx))
-        entries.append(cmake_cache_path("MPI_Fortran_COMPILER", spec["mpi"].mpifc))
+
+        # not all MPIs have Fortran wrappers
+        if hasattr(spec["mpi"], "mpifc"):
```

Comment:
I don't know the Spack style, but I might expect a single attribute reference with
```py
try: f = spec["mpi"].mpifc
except AttributeError: pass  # not all MPIs have Fortran wrappers
else: entries.append(cmake_cache_path("MPI_Fortran_COMPILER", f))
```

Reply:
I'd leave it the Spack devs to state what style they prefer.